services/calibration_service.py

```diff
@@ -60,163 +61,6 @@
-    # names contain the transit pass time and calibrator names.
-    yesterday = (ttime-1*u.d).isot.split('T')[0]
-    filenames_yesterday = get_files_for_cal(
-        caltable,
-        REFCORR,
-        CALTIME,
-        FILELENGTH,
-        hdf5dir=HDF5DIR,
-        date_specifier='{0}*'.format(yesterday),
-    )
-    if yesterday in filenames_yesterday.keys():
-        filenames[yesterday] = filenames_yesterday[yesterday]
-    else:
-        filenames[yesterday] = {}
-    # Get rid of calibrators after the snap start time or without files
-    for date in filenames.keys():
-        for cal in list(filenames[date].keys()):
-            if filenames[date][cal]['transit_time'] < start_time or \
-                len(filenames[date][cal]['files'])==0 or \
-                filenames[date][cal]['transit_time'] > ttime:
-                filenames[date].pop(cal)
-    # Sort the filenames by time
-    assert len(filenames.keys()) < 3
-    filenames = sort_filenames(filenames)
-    # Average beamformer solutions
-    beamformer_names = []
-    for date in filenames.keys():
-        for cal in filenames[date].keys():
-            cal_ttime = filenames[date][cal]['transit_time']
-            cal_ttime.precision = 0
-            beamformer_names += [
-                '{0}_{1}'.format(
-                    cal,
-                    cal_ttime.isot
-                )
-            ]
-    # Open yaml files
-    print('opening yaml files')
-    if os.path.exists(
-        '{0}/beamformer_weights_{1}.yaml'.format(
-            BEAMFORMER_DIR,
-            beamformer_names[0]
-        )
-    ):
-        with open(
-            '{0}/beamformer_weights_{1}.yaml'.format(
-                BEAMFORMER_DIR,
-                beamformer_names[0]
-            )
-        ) as f:
-            latest_solns = yaml.load(f, Loader=yaml.FullLoader)
-        for bfname in beamformer_names[1:].copy():
-            try:
-                with open(
-                    '{0}/beamformer_weights_{1}.yaml'.format(
-                        BEAMFORMER_DIR,
-                        bfname
-                    )
-                ) as f:
-                    solns = yaml.load(f, Loader=yaml.FullLoader)
-                assert solns['cal_solutions']['antenna_order'] == \
-                    latest_solns['cal_solutions']['antenna_order']
-                assert solns['cal_solutions']['corr_order'] == \
-                    latest_solns['cal_solutions']['corr_order']
-                assert solns['cal_solutions']['delays'] == \
-                    latest_solns['cal_solutions']['delays']
-                assert solns['cal_solutions']['eastings'] == \
-                    latest_solns['cal_solutions']['eastings']
-            except (AssertionError, FileNotFoundError):
-                beamformer_names.remove(bfname)
-    else:
-        beamformer_names = []
-        latest_solns = None
-    return beamformer_names, latest_solns
```

Comment:
Replaced by look-up of all recent solutions (not just ones at the current dec) in weights.py